include/dfrn.php

```diff
@@ -1045,4 +1067,1343 @@ function deliver($owner,$contact,$atom, $dissolve = false) {
+						$author_remove = (($item["origin"] && $item["self"]) ? true : false);
+						$author_copy = (($item["origin"]) ? true : false);
+
+						if($owner_remove && $author_copy)
+							return;
+						if($author_remove || $owner_remove) {
+							$tags = explode(',',$i[0]["tag"]);
+							$newtags = array();
+							if(count($tags)) {
+								foreach($tags as $tag)
+									if(trim($tag) !== trim($xo->body))
+										$newtags[] = trim($tag);
+							}
+							q("UPDATE `item` SET `tag` = '%s' WHERE `id` = %d",
+								dbesc(implode(',',$newtags)),
+								intval($i[0]["id"])
+							);
+							create_tags_from_item($i[0]["id"]);
+						}
+					}
+				}
+			}
+
+			if($entrytype == DFRN_TOP_LEVEL) {
+				$r = q("UPDATE `item` SET `deleted` = 1, `edited` = '%s', `changed` = '%s',
+						`body` = '', `title` = ''
+					WHERE `parent-uri` = '%s' AND `uid` = %d",
+						dbesc($when),
+						dbesc(datetime_convert()),
+						dbesc($uri),
+						intval($importer["uid"])
+					);
+				create_tags_from_itemuri($uri, $importer["uid"]);
+				create_files_from_itemuri($uri, $importer["uid"]);
+				update_thread_uri($uri, $importer["uid"]);
+			} else {
+				$r = q("UPDATE `item` SET `deleted` = 1, `edited` = '%s', `changed` = '%s',
+						`body` = '', `title` = ''
+					WHERE `uri` = '%s' AND `uid` = %d",
+						dbesc($when),
+						dbesc(datetime_convert()),
+						dbesc($uri),
+						intval($importer["uid"])
+					);
+				create_tags_from_itemuri($uri, $importer["uid"]);
+				create_files_from_itemuri($uri, $importer["uid"]);
+				update_thread_uri($uri, $importer["importer_uid"]);
+				if($item["last-child"]) {
+					// ensure that last-child is set in case the comment that had it just got wiped.
+					q("UPDATE `item` SET `last-child` = 0, `changed` = '%s' WHERE `parent-uri` = '%s' AND `uid` = %d ",
+						dbesc(datetime_convert()),
+						dbesc($item["parent-uri"]),
+						intval($item["uid"])
+					);
+					// who is the last child now?
+					$r = q("SELECT `id` FROM `item` WHERE `parent-uri` = '%s' AND `type` != 'activity' AND `deleted` = 0 AND `moderated` = 0 AND `uid` = %d
+						ORDER BY `created` DESC LIMIT 1",
+							dbesc($item["parent-uri"]),
+							intval($importer["uid"])
+					);
+					if(count($r)) {
+						q("UPDATE `item` SET `last-child` = 1 WHERE `id` = %d",
+							intval($r[0]["id"])
+						);
+					}
+				}
+				// if this is a relayed delete, propagate it to other recipients
+
+				if($entrytype == DFRN_REPLY_RC) {
+					logger("Notifying followers about deletion of post ".$item["id"], LOGGER_DEBUG);
+					proc_run("php", "include/notifier.php","drop", $item["id"]);
+				}
+			}
+		}
+	}
+
+	/**
+	 * @brief Imports a DFRN message
+	 *
+	 * @param text $xml The DFRN message
+	 * @param array $importer Record of the importer user mixed with contact of the content
+	 * @param bool $sort_by_date Is used when feeds are polled
+	 */
+	function import($xml,$importer, $sort_by_date = false) {
```

Comment:
I have talked to @fabrixxm according the explicit declaration of public static function
Can you change it in
```
public static function import($xml,$importer, $sort_by_date = false) {
```
?